File: src/pages/exercises/string-citation/_codes/js/response/citation.js
function citation(name) {
  let result = '';
  let words = name.split(' ');
  let lastWord = words.pop();
  lastWord = lastWord.toUpperCase();
  words = words.join(' ');
  result = `${lastWord}; ${words}`;
  return result;
}

function compactCitation(name) {
  let result = '';
  let words = name.split(' ');
  let lastWord = words.pop();
  lastWord = lastWord.toUpperCase();
  // for(let index in words){
  //   words[index] = `${words[index][0].toUpperCase()}.`
  // }
  words = words.map(word => `${word[0].toUpperCase()}.`);
  words = words.join(' ');
  result = `${lastWord}; ${words}`;
  return result;
}

export { citation, compactCitation };

File: src/pages/exercises/string-citation/_codes/ruby/response/src/citation.rb
class Citation
    def Citation.longCitation(name)
        result = ''
        words = name.split(' ')
        lastWord = words.pop()
        lastWord = lastWord.upcase
        words = words.join(' ')
        result = "#{lastWord}; #{words}"
        return result
    end

    def Citation.compactCitation(name)
        result = ''
        words = name.split(' ')
        lastWord = words.pop()
        lastWord = lastWord.upcase
        words = words.map{ |word| word[0].upcase + '.' }
        words = words.join(' ')
        result = "#{lastWord}; #{words}"
        return result
    end
end